Source/WebCore/rendering/svg/SVGTextLayoutEngineSpacing.cpp

```diff
@@ -34,20 +34,19 @@ SVGTextLayoutEngineSpacing::SVGTextLayoutEngineSpacing(const FontCascade& font)
 {
 }
 
-float SVGTextLayoutEngineSpacing::calculateCSSSpacing(const char16_t* currentCharacter)
+float SVGTextLayoutEngineSpacing::calculateCSSSpacing(std::optional<char16_t> currentCharacter)
 {
-    const char16_t* lastCharacter = m_lastCharacter;
+    std::optional<char16_t> lastCharacter = m_lastCharacter;
     m_lastCharacter = currentCharacter;
 
     if (!m_font->letterSpacing() && !m_font->wordSpacing())
         return 0;
 
     float spacing = m_font->letterSpacing();
-    if (currentCharacter && lastCharacter && m_font->wordSpacing()) {
-        if (FontCascade::treatAsSpace(*currentCharacter) && !FontCascade::treatAsSpace(*lastCharacter))
+    if (m_font->wordSpacing()) {
+        if (currentCharacter && FontCascade::treatAsSpace(*currentCharacter) && (!lastCharacter || !FontCascade::treatAsSpace(*lastCharacter)))
```

Comment:
This code has repetition:
```
auto needsWordSpacing = [](std::optional<char16_t> ch) {
    return ch && FontCascade::treatAsSpace(*ch));
};

if (m_font->wordSpacing() && needsWordSpacing(currentCharacter) && !needsWordSpacing(lastCharacter))
    spacing += m_font->wordSpacing();
```

Reply:
Updated. Tried to address all feedback. @shallawa